src/main/java/uk/gov/register/filters/DeprecatedFormatFilter.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.register.filters;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import java.net.URI;
+
+import com.google.common.net.HttpHeaders;
+import uk.gov.register.views.representations.ExtraMediaType;
+
+@Provider
+public class DeprecatedFormatFilter implements ContainerResponseFilter {
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+        final MultivaluedMap<String, Object> headers = responseContext.getHeaders();
+        final MediaType mediaType;
+
+        try {
+            mediaType = (MediaType) headers.getFirst(HttpHeaders.CONTENT_TYPE);
+        } catch (ClassCastException e) {
+            return;
+        }
+
+        if (mediaType == null) {
+            return;
+        }
+
+        final UriInfo uriInfo = requestContext.getUriInfo();
+        final URI requestedUri = uriInfo.getRequestUri();
+        final String alternate;
+
+        if (recommendJson(mediaType)) {
+            alternate = alternateLink(requestedUri, ".json");
+        } else if (recommendCsv(mediaType)) {
+            alternate = alternateLink(requestedUri, ".csv");
+        } else {
+            return;
+        }
+
+        addWarning(headers, mediaType.getSubtype(), alternate);
+    }
+
+    private String alternateLink(URI requestedUri, String recommendedExtension) {
+        final String path = requestedUri.getPath();
+        final UriBuilder builder = UriBuilder.fromUri(requestedUri);
+
+        // Even if an extension was specified in the original URI, it has been stripped out
+        // at this point, so just add the new one (before any trailing slash).
+        final String newPath = path.replaceFirst("(/)?$", recommendedExtension + "$1");
+
+        builder.replacePath(newPath);
+
+        return builder.toTemplate();
+    }
+
+    private void addWarning(MultivaluedMap<String, Object> headers, String deprecatedType, String alternate) {
+        headers.add(HttpHeaders.LINK, alternate + "; rel=\"alternate\"");
+        headers.add(HttpHeaders.WARNING, "299 - \"Miscellaneous Persistent Warning\" \"" + deprecatedType + " is deprecated and will be removed\"");
```

Comment:
Perhaps we could extend this message suggesting they check the alternate header for a suitable alternative?